server/apps/distributed-app/docs/modules/ROOT/pages/configure/ssl.adoc

```diff
@@ -84,6 +84,32 @@ Please note `JKS` keystore format is also supported (default value if no keystor
 </tls>
 ....
 
+
+=== Client authentication via certificates
+
+When you enable TLS, you may also configure the server to request a client certificate for authentication:
+
+....
+<tls socketTLS="false" startTLS="true">
+  <keystore>file://conf/keystore</keystore>
+  <keystoreType>JKS</keystoreType>
+  <secret>yoursecret</secret>
+
+  <clientAuth required="true">
+    <truststore>file://conf/truststore</truststore>
+    <truststoreType>JKS</truststoreType>
+    <truststoreSecret>yoursecret</truststoreSecret>
+  </clientAuth>
+</tls>
+....
+
+With `required="true"`, the server refuses any client connections that do not include a client certificate.
+
+With `required="false"`, the servers requests an optional client certificate, but also accepts connections without one.
```

Comment:
What is the benefit of `required=false` ?

Reply:
It is one of the options you have in the TLS handshake protocol. The idea is that servers could allow different levels of access whether they decided to authenticate or not.
Right now this does not make much sense in James. But someone could implement a way to access the client certificate (if any) from the protocol session, e.g. to implement SASL EXTERNAL authentication mode. Or even from mailets to do whatever.
Now that I think about it, it is probably easier to just remove the option for now; if someone needs it later they can easily add it again.

Reply:
Fixed in latest push. I decided to keep the three states in an enum because that's how TLS works, but removed the option for WANT from the configuration. Now its either NEED with `<clientAuth>...</clientAuth>` ot NONE without this XML element.